### PR TITLE
Feature/cloudtrail cloudwatch log

### PR DESCRIPTION
**Pull request by Djeeteg007**
This features makes it possible to enable Cloudwatch Log in cloudtrail. This could be useful for attaching Cloudwatch Alarms, which can be triggered when some unwanted cloudwatch API event occurs.

### CODE REVIEW

**Comment by npalm** [comment]
@JoostSaanen Thanks, will check your contribution in the next days

File: policies/cloudwatch_policy.tpl [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogStream"
+      ],
+      "Resource": [
+        "${log_group_arn}"
```
Comment (gertjanmaas) [inline policies/cloudwatch_policy.tpl]: Doesn't this need to be `"${log_group_arn}*"` to allow creation of log streams inside that log group?

File: policies/cloudwatch_policy.tpl [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:PutLogEvents"
+      ],
+      "Resource": [
+        "${log_group_arn}"
```
Comment (gertjanmaas) [inline policies/cloudwatch_policy.tpl]: Doesn't this need to be `"${log_group_arn}*"` to allow events to be put inside streams in that log group?

**Review by gertjanmaas — CHANGES_REQUESTED** [review]

File: cloudtrail.tf [hunk 3]
```diff
@@ -20,6 +22,48 @@ resource "aws_cloudtrail" "cloudtrail" {
...
+#
+# CloudTrail Cloudwatch IAM Role
+#
+data "template_file" "cloudwatch_iam_assume_role_policy_document" {
+  template = "${file("${path.module}/policies/cloudwatch_assume_role_policy.tpl")}"
+}
+
+resource "aws_iam_role" "cloudwatch_iam_role" {
+  count              = "${var.enable_cloudwatch_logs}"
+  name               = "terraform-cloudwatch-role"
```
Comment (npalm) [inline cloudtrail.tf]: Please extract this to a variable and choose a default.

File: cloudtrail.tf [hunk 4]
```diff
@@ -20,6 +22,48 @@ resource "aws_cloudtrail" "cloudtrail" {
...
+  template = "${file("${path.module}/policies/cloudwatch_policy.tpl")}"
+
+  vars {
+    log_group_arn = "${aws_cloudwatch_log_group.log_group.arn}"
+  }
+}
+
+resource "aws_iam_policy" "cloudwatch_iam_policy" {
+  count  = "${var.enable_cloudwatch_logs}"
+  name   = "terraform-cloudwatch-policy"
```
Comment (npalm) [inline cloudtrail.tf]: Same here, extract to a variable, with default

File: examples/cloudtrail-with-cloudwatch-log/main.tf [hunk 5]
```diff
@@ -0,0 +1,16 @@
+provider "aws" {
+  region  = "eu-west-1"
+  version = "1.24.0"
```
Comment (npalm) [inline examples/cloudtrail-with-cloudwatch-log/main.tf]: Please can you update the provider to the latest one? 2.x

File: examples/cloudtrail-with-cloudwatch-log/main.tf [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+provider "template" {
+  version = "1.0"
```
Comment (npalm) [inline examples/cloudtrail-with-cloudwatch-log/main.tf]: Please can you update the provide to the latest one

**Review by npalm — CHANGES_REQUESTED** [review]
@JoostSaanen thanks for the PR, please can you have a look at the comments of @gertjanmaas and @npalm

File: cloudtrail.tf [hunk 7]
```diff
@@ -20,6 +22,48 @@ resource "aws_cloudtrail" "cloudtrail" {
   tags = "${var.tags}"
 }
 
+#
+# CloudTrail Cloudwatch Log Group
+#
+resource "aws_cloudwatch_log_group" "log_group" {
+  count = "${var.enable_cloudwatch_logs}"
+  name = "CloudTrail/DefaultLogGroup"
```
[the remark this answers is not in the document]
Comment (JoostSaanen) [reply cloudtrail.tf]: will do

File: cloudtrail.tf (hunk 4 above)
Comment (JoostSaanen) [reply cloudtrail.tf]: Same for this one: `terraform-awsconfig-policy`.

Anyway, I'll change it for al configurable stuff in the scope of this PR.

**Review by npalm — APPROVED** [review]

**Pull request merged**